src/MergeContext.js

```diff
@@ -108,6 +108,84 @@ class Approval {
     }
 }
 
+class StatusCheck
+{
+    constructor(raw) {
+        assert(raw.context);
+        assert(raw.state);
+        assert(raw.target_url);
+        assert(raw.description);
+
+        this.context = raw.context;
+        this.state = raw.state;
+        this.targetUrl = raw.target_url;
+        this.description = raw.description;
+
+        this._raw = raw;
+    }
+}
+
+// aggregates (required) status check for a PR or commit
+class StatusChecks
+{
+    // expectedStatusCount:
+    //   for staged commits: the bot-configured number of required checks (Config.stagingChecks()),
+    //   for PR commits: GitHub configured number of required checks (requested from GitHub)
+    // context: either "PR" or "Staging";
+    constructor(expectedStatusCount, context) {
+        assert(expectedStatusCount !== undefined);
+        assert(expectedStatusCount !== null);
+        assert(context);
+        this.expectedStatusCount = expectedStatusCount;
+        this.context = context;
+        this.requiredStatuses = [];
+    }
+
+    addRequiredStatus(requiredStatus) {
+        assert(requiredStatus);
+        this.requiredStatuses.push(requiredStatus);
+    }
+
+    // no more required status changes or additions are expected
+    final() {
+        return (this.requiredStatuses.length >= this.expectedStatusCount) &&
+            this.requiredStatuses.every(check => check.state !== 'pending');
+    }
+
+    // something went wrong with at least one of the required status checks
+    failed() {
+        return this.requiredStatuses.some(check => check.state !== 'pending' && check.state !== 'success');
+    }
+
+    // the results are final and all checks were a success
+    succeeded() {
+        return this.final() && this.requiredStatuses.every(check => check.state === 'success');
+    }
+
+    toString() {
+        let combinedStatus = "context: " + this.context + " expected/received: " + this.expectedStatusCount + "/" +
+            this.requiredStatuses.length + ", combined: ";
+        if (this.failed())
+            combinedStatus += "failure";
+        else if (this.succeeded())
+            combinedStatus += "success";
+        else if (this.final()) {
+            // final() implies failed() or succeeded()
```

Comment:
Let's move this comment so that it trails the assignment below it. That way, we can remove curly braces and all these chained `if` statements will look similar (as they should).

Reply:
Done.